### PR TITLE
Clarify Typing.

### PR DESCRIPTION
**Pull request by Carreau**
From what I can tell, those are always path-or-paths from the types.

From the implementation that may vary. I think it's actually never Path,
but always str.

### CODE REVIEW

**Comment by tlambert03** [comment]
Yeah, I've been wanting to clarify this too.  We _might_ need to double check this on the napari side, to make sure that all path objects have indeed been coerced to string before we call out to npe2.  but this is definitely how I think it should be (no path objects by this point)

**Comment by Carreau** [comment]
How would you feel about not trying to be a bit stricter about what we pass to read ?
I thinks that either one or a list of either strings or path is a 4 item matrix that starts to get in the way of readability and consistency. 

I think that having npe2 always normalising to a list of string would be beneficial. Unless there is a semantic difference between a (list of 1 item) and a single item on it's own, I both believe that it would benefit everyone.

**Comment by Carreau** [comment]
> Yeah, I've been wanting to clarify this too. We _might_ need to double check this on the napari side, to make sure that all path objects have indeed been coerced to string before we call out to npe2. but this is definitely how I think it should be (no path objects by this point)

Ok, then I think we should start putting assert/warnings everywhere.

**Comment by tlambert03** [comment]
ok!  those are real cheap right?

**Comment by tlambert03** [comment]
lol.  fair point :)

"OMGZ ... WHAT ABOUT THE PERFORMANCE IMPACT??"

**Comment by Carreau** [comment]
> WHAT ABOUT THE PERFORMANCE IMPACT

Switch to Julia.

**Comment by tlambert03** [comment]
> And what do you think about always calling the contributions with a list of path ?

unfortunately, that was claimed semantically early on to mean "please stack these paths yourself".  Not that I like that pattern... but we can't just change that without a lot of communication

**Comment by Carreau** [comment]
And how many users current use npe2 readers contributions ? Would the migration to npe2 be a good time to do this ?

**Comment by tlambert03** [comment]
conceivably... but it's a bigger conversation.  I'm more than happy to have it though!  We'd need to add a `stack=True` argument to both the get_reader and the reader i guess, which means we either need to break API or do inspection.  All doable, just some decisions to be made and needs a clear plan

File: npe2/io_utils.py [hunk 1]
```diff
@@ -38,9 +38,10 @@ def read(path: PathLike, *, plugin_name: Optional[str] = None) -> List[LayerData
 
 
 def read_get_reader(
-    path: PathLike, *, plugin_name: Optional[str] = None
+    path: Union[str, Sequence[str]], *, plugin_name: Optional[str] = None
 ) -> Tuple[List[LayerData], ReaderContribution]:
     """Variant of `read` that also returns the `ReaderContribution` used."""
+    _ensure_str_or_seq_str(path)
```
Comment (tlambert03) [inline npe2/io_utils.py:44]: this (and `read`) is actually one place where I think we should allow Path objects.  This is part of the public api that someone (besides napari) can use to take advantage of napari plugins without having to open up napari.  In other words, these functions are there to satisfy the requests in https://github.com/napari/napari-plugin-engine/issues/24

Comment (Carreau) [reply npe2/io_utils.py:44]: I'm happy to cast here, though from a [github search](https://github.com/search?q=read_get_reader&type=code) it's only used on a test Repo of Juan, so I _think_ we still have the opportunity to tell users that they should pass in strings.

**Review by tlambert03 — APPROVED** [review]
this looks good to me @Carreau.  is it ready to go?  I can go either way on the side conversation about the public `npe2.read` signature

**Pull request merged**